Packages/Core/Runtime/StackTrace/StackTraceEntry.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using UnityEngine;
+
+namespace UnityAtoms
+{
+    public class StackTraceEntry : IEquatable<StackTraceEntry>
+    {
+        private readonly Guid _id;
+        private readonly int _frameCount;
+        private readonly string _stackTrace;
+        private readonly object _value;
+        private readonly bool _constructedWithValue = false;
+
+        private StackTraceEntry(string stackTrace)
+        {
+            _id = Guid.NewGuid();
+            _stackTrace = stackTrace;
+
+            if (Application.isPlaying)
+            {
+                _frameCount = Time.frameCount;
+            }
+        }
+        private StackTraceEntry(string stackTrace, object value)
+        {
+            _value = value;
+            _constructedWithValue = true;
+            _id = Guid.NewGuid();
+            _stackTrace = stackTrace;
+
+            if (Application.isPlaying)
+            {
+                _frameCount = Time.frameCount;
+            }
+        }
+
+        public static StackTraceEntry Create(object obj) => new StackTraceEntry(Environment.StackTrace, obj);
+
+        public static StackTraceEntry Create() => new StackTraceEntry(Environment.StackTrace);
```

Comment:
probably better use: `new StacktTrace(4).ToString()` (will skip 4 frames, which are always: 1. get_StackTrace() / or cctor, 2. this line itself, 3. EventBase:Raise(), 4. Event.Raise(T item)`

this might be a different frame count for both Create methods, but it will Atom internal stuff, which is always the same

Reply:
Fixed. Also added skipFrames as a parameter to the `Create` method. Since the StackTrace is created here in this method the correct amount of stack traces to shave off was 3 instead of 4.